Fix SqlValue sort_by_key usage in executor tests (requires Ord trait)

## Problem

Discovered while working on #22: The executor tests fail to compile because `select_executor.rs:468` uses `sort_by_key` on `SqlValue`, which requires the `Ord` trait, but only `PartialOrd` was implemented in PR #32.

## Error

```
error[E0277]: the trait bound `SqlValue: Ord` is not satisfied
   --> crates/executor/src/tests/select_executor.rs:468:13
    |
468 |     results.sort_by_key(|(dept, _)| dept.clone());
    |             ^^^^^^^^^^^ the trait `Ord` is not implemented for `SqlValue`
```

## Root Cause

- Issue #16 was titled "Implement PartialOrd/**Ord**" but PR #32 only implemented `PartialOrd`
- The test at line 468 was written assuming `Ord` would be available
- This blocks documentation work (#22) that needs to run `cargo coverage`

## Proposed Solution

**Option 1: Implement Ord trait** (if SQL semantics allow total ordering)
- Add `impl Ord for SqlValue` in `crates/types/src/lib.rs`
- Handle NULL and type mismatches appropriately
- Verify SQL:1999 compliance

**Option 2: Fix the test** (if Ord is not appropriate)
- Change line 468 from `sort_by_key` to `sort_by` with custom comparator
- Match the pattern used elsewhere in the codebase
- Similar to how PR #15 fixed other sorting issues

## Impact

**Blocks**:
- #22: Cannot verify coverage commands work
- Any other work requiring `cargo test` or `cargo coverage` to pass

**Files Affected**:
- `crates/executor/src/tests/select_executor.rs:468`
- Possibly `crates/types/src/lib.rs` if implementing Ord

## Context

Discovered during Builder work on #22 (Phase 2: Update technical documentation). The Curator's implementation guidance requires running `cargo coverage` as Step 1, but this compilation error prevents that.

This appears to be the same "pre-existing error" mentioned in PR #32's description but was never tracked as a separate issue.

--- a/crates/executor/src/tests/select_executor.rs
+++ b/crates/executor/src/tests/select_executor.rs
@@ -465,7 +465,9 @@ fn test_group_by_with_count() {
         .into_iter()
         .map(|row| (row.values[0].clone(), row.values[1].clone()))
         .collect::<Vec<_>>();
-    results.sort_by_key(|(dept, _)| dept.clone());
+    results.sort_by(|(a_dept, _), (b_dept, _)| {
+        a_dept.partial_cmp(b_dept).unwrap_or(std::cmp::Ordering::Equal)
+    });
     assert_eq!(results[0], (types::SqlValue::Integer(1), types::SqlValue::Integer(2)));
     assert_eq!(results[1], (types::SqlValue::Integer(2), types::SqlValue::Integer(1)));
 }

